### PR TITLE
feat: use `getset` lib for deriving getters and setters

### LINKED ISSUE #87: Use getset library for deriving getters and setters (https://crates.io/crates/getset)

### PR DESCRIPTION
**Pull request by duguorong009**
## Description
- Use the `getset` crate for deriving getters and setters of structs in codebase

## Related issues
- Resolves #87 

## Changes
- Update the derive of structs in `openrank-common` pkg, in order to use `getset` pkg

### CODE REVIEW

File: common/src/tx/mod.rs [hunk 1]
```diff
@@ -101,15 +102,21 @@ impl Body {
...
+    #[getset(get = "pub")]
     // Use 0x0 for transactions intended to be processed by the network
     to: Address,
+    #[getset(skip)]
     body: Body,
+    #[getset(get = "pub")]
     signature: Signature,
+    #[getset(skip)]
     sequence_number: Option<u64>,
 }
```
Comment (lazovicff) [inline common/src/tx/mod.rs:121]: To simplify things, lets just put #[getset(get = "pub")] on top of Tx, so it applies to all fields.
```suggestion
#[derive(Debug, Clone, PartialEq, RlpEncodable, RlpDecodable, Serialize, Deserialize, Getters)]
#[rlp(trailing)]
#[getset(get = "pub")]
pub struct Tx {
    nonce: u64,
    from: Address,
    // Use 0x0 for transactions intended to be processed by the network
    to: Address,
    body: Body,
    signature: Signature,
    sequence_number: Option<u64>,
}
```

The code should be updated to accommodate for this change.

Same applies to everywhere we use Getters.
For Setters, individual fields should be picked and made public, for safety reasons

File: common/src/tx/trust.rs [hunk 2]
```diff
@@ -132,9 +133,13 @@ impl Decodable for TrustEntry {
     }
 }
 
+#[derive(Getters)]
 pub struct AcceptedTrustUpdates {
+    #[get = "pub with_prefix"]
```
Comment (lazovicff) [inline common/src/tx/trust.rs:138]: We should be consistent. Either we use prefix for all or none

File: common/src/tx/mod.rs (hunk 1 above)
Comment (duguorong009) [reply common/src/tx/mod.rs]: Done 1ccfa54

File: common/src/tx/trust.rs (hunk 2 above)
Comment (duguorong009) [reply common/src/tx/trust.rs]: Done 1ccfa54

**Comment by lazovicff** [comment]
All the structs in the codebase that have `pub` inside:
- For structs in ./common - Remove `pub` and implement getters
- For structs inside local nodes - Remove `pub`

**Comment by lazovicff** [comment]
- Make MyBehaviour fields private:
https://github.com/openrankprotocol/openrank/blob/use-getset-lib/common/src/lib.rs#L29-L32
Make proxy methods to call underlying methods for gossipsub or mdns
- Make inner fields private for Hash, TxHash, OwnedNamespace

**Review by lazovicff — APPROVED** [review]

**Pull request merged**